dataflow/src/main/java/org/checkerframework/dataflow/reachdefinition/ReachDefinitionStore.java

```diff
@@ -0,0 +1,116 @@
+package org.checkerframework.dataflow.reachdefinition;
+
+import org.checkerframework.checker.nullness.qual.Nullable;
+import org.checkerframework.dataflow.analysis.Store;
+import org.checkerframework.dataflow.cfg.visualize.CFGVisualizer;
+import org.checkerframework.dataflow.expression.JavaExpression;
+import org.checkerframework.javacutil.BugInCF;
+
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.LinkedHashSet;
+import java.util.Set;
+import java.util.StringJoiner;
+
+/** A reach definition store contains a set of reach definitions represented by nodes. */
+public class ReachDefinitionStore implements Store<ReachDefinitionStore> {
+
+    /** A set of reach definitions abstract values. */
+    private final Set<ReachDefinitionValue> reachDefSet;
+
+    /** Create a new ReachDefinitionStore. */
+    public ReachDefinitionStore() {
+        reachDefSet = new LinkedHashSet<>();
+    }
+
+    /**
+     * Create a new ReachDefinitionStore.
+     *
+     * @param reachDefSet a set of reach definition abstract values
+     */
+    public ReachDefinitionStore(Set<ReachDefinitionValue> reachDefSet) {
+        this.reachDefSet = reachDefSet;
+    }
+
+    /**
+     * Remove the information of a reach definition from the reach definition set.
+     *
+     * @param def a reach definition
+     */
+    public void killDef(ReachDefinitionValue def) {
+        Iterator<ReachDefinitionValue> it = reachDefSet.iterator();
+        while (it.hasNext()) {
+            ReachDefinitionValue existedDef = it.next();
+            if (existedDef.defs.getTarget().toString().equals(def.defs.getTarget().toString())) {
```

Comment:
By passing a node to the assignment target to this method, you can simply compare `existedDef.defs.getTarget().equals(target)` here.

Reply:
Got it.